### DISCUSSION TITLE
Doric emits LED even pyBoard is not connected to computer

### DISCUSSION THREAD

**Discussion by TetsufumiIto** [pyPhotometry/code#14] [Q&A]
Hi all, this is the first time for me to post.

I purchased the pyPhotometry from OpenEphys around the end of October of last year, and started testing the system in these days, but it does not work.

When my pyboard is connected to Doric photometry system (LED out to Doric E port, and Doric F port to ANALOG1 input), Doric emits blue light (see my right hand in 1.jpg) even the pyBoard is not connected with computer, meaning that the power of pyboard is supplied from ANALOG1 input. 

![1](https://github.com/pyPhotometry/code/assets/162299737/7712fe82-82f8-4188-a2d7-0fab5fd826ef)

Indeed, when I disconnected the ANALOG1 input, LED emittion stopped (3.jpg). 

![3](https://github.com/pyPhotometry/code/assets/162299737/d3d2eba5-93b1-459f-a33a-da81ab92b0fa)

When I connect LED out to oscilloscope,big 50Hz big voltage is recorded (5.jpg). 

![5](https://github.com/pyPhotometry/code/assets/162299737/d0fbf2f2-a486-4025-9831-4cbddf0590c2)

Touching BNC ground to ANALOG1 ground is enough to emit the LED.

This phenomena continue even when I connect the pyboard to computer. At this condition, pyboard is recognized by computer, and pyphotometry program detected the board (but signal is always saturated and LED is always on). 

I will be really appreciated if you give me any suggestions.

**Comment by ThomasAkam** [comment]
Hi @TetsufumiIto,

Is the Doric minicube one of the generation 3 minicubes that have built in LED drivers and LEDs or the generation 2 minicubes that have built in LEDs but not LED drivers?  The pyPhotometry board has LED drivers on the board so the LED outputs should be connected directly to LEDs not to the control inputs of LED drivers.  This means the board is compatible with the Doric generation 2 minicubes but not the generation 3 minicubes.  

If you have a generation 3 minicube you could adapt pyPhotometry to work with it by bypassing the LED drivers on the pyPhotometry board by connecting the analog output pins on the pyboard that control the led current directly to the inputs of the led drivers on the minicube.

**Reply by TetsufumiIto** [reply]
Thank you for the reply. Yes. This is Gen3 (iLFMC4-G3), so as you noted, we have to bypass the driver circuit. The pins which I have to connect are X5 and GND for LED1, and X6 and GND for LED2, is this understanding correct?
Tetsufumi

**Reply by ThomasAkam** [reply]
Yes, that is correct.  The easiest way to make the connection might be to solder wires onto the bottom of the pyPhotometry board where the pins from the pyboard come through.
